iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/historical/PipeHistoricalDataRegionTsFileAndDeletionExtractor.java

```diff
@@ -651,6 +721,42 @@ public synchronized Event supply() {
     }
   }
 
+  private Event supplyDeletionEvent(final DeletionResource deletionResource) {
+    final PipeDeleteDataNodeEvent event =
+        new PipeDeleteDataNodeEvent(
+            deletionResource.getDeleteDataNode(),
+            pipeName,
+            creationTime,
+            pipeTaskMeta,
+            treePattern,
+            tablePattern,
+            false);
+    if (sloppyPattern || isDbNameCoveredByPattern) {
+      event.skipParsingPattern();
+    }
+    if (sloppyTimeRange) {
+      event.skipParsingTime();
+    }
+
+    final boolean isReferenceCountIncreased =
+        event.increaseReferenceCount(
+            PipeHistoricalDataRegionTsFileAndDeletionExtractor.class.getName());
+    if (!isReferenceCountIncreased) {
+      LOGGER.warn(
+          "Pipe {}@{}: failed to increase reference count for historical deletion event {}, will discard it",
+          pipeName,
+          dataRegionId,
+          event);
+    } else {
+      Optional.ofNullable(DeletionResourceManager.getInstance(String.valueOf(dataRegionId)))
+          .ifPresent(
+              manager ->
+                  event.setDeletionResource(
+                      manager.increaseResourceReferenceAndGet(event.getDeleteDataNode())));
```

Comment:
This may delete the resource directly when some of the tasks have finished but others have not extracted it. This may be a huge complicated problem....

Reply:
fixed. deletionResource's referenceCount is set to replicateNums initially